From patchwork Fri Feb 22 21:06:43 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: kbuild test robot <lkp@intel.com>
X-Patchwork-Id: 10827003
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1A7C4180E
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 22 Feb 2019 21:07:03 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 01F5932CCB
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 22 Feb 2019 21:07:03 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E9EE932CCF; Fri, 22 Feb 2019 21:07:02 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id CFF0532CCB
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 22 Feb 2019 21:07:01 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726298AbfBVVHA (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 22 Feb 2019 16:07:00 -0500
Received: from mga05.intel.com ([192.55.52.43]:15926 "EHLO mga05.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1726152AbfBVVG7 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 22 Feb 2019 16:06:59 -0500
X-Amp-Result: UNKNOWN
X-Amp-Original-Verdict: FILE UNKNOWN
X-Amp-File-Uploaded: False
Received: from orsmga007.jf.intel.com ([10.7.209.58])
  by fmsmga105.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 22 Feb 2019 13:06:59 -0800
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.58,401,1544515200";
   d="scan'208";a="117086687"
Received: from lkp-server01.sh.intel.com (HELO lkp-server01) ([10.239.97.150])
  by orsmga007.jf.intel.com with ESMTP; 22 Feb 2019 13:06:57 -0800
Received: from kbuild by lkp-server01 with local (Exim 4.89)
        (envelope-from <lkp@intel.com>)
        id 1gxI25-0004gO-56; Sat, 23 Feb 2019 05:06:57 +0800
Date: Sat, 23 Feb 2019 05:06:43 +0800
From: kbuild test robot <lkp@intel.com>
To: "Singh, Brijesh" <brijesh.singh@amd.com>
Cc: kbuild-all@01.org, kvm@vger.kernel.org,
        Robert Hu <robert.hu@intel.com>,
        Farrah Chen <farrah.chen@intel.com>,
        Danmei Wei <danmei.wei@intel.com>,
        Paolo Bonzini <pbonzini@redhat.com>
Subject: [PATCH] KVM: SVM: fix boolreturn.cocci warnings
Message-ID: <20190222210643.GA12790@lkp-ib03>
References: <201902230537.rg4og2Ma%fengguang.wu@intel.com>
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
In-Reply-To: <201902230537.rg4og2Ma%fengguang.wu@intel.com>
User-Agent: Mutt/1.5.23 (2014-03-12)
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: kbuild test robot <lkp@intel.com>

arch/x86/kvm/vmx/vmx.c:7413:8-9: WARNING: return of 0/1 in function 'vmx_need_emulation_on_page_fault' with return type bool

 Return statements in functions returning bool should use
 true/false instead of 1/0.
Generated by: scripts/coccinelle/misc/boolreturn.cocci

Fixes: 4589eecce385 ("KVM: SVM: Workaround errata#1096 (insn_len maybe zero on SMAP violation)")
CC: Singh, Brijesh <brijesh.singh@amd.com>
Signed-off-by: kbuild test robot <fengguang.wu@intel.com>
---

tree:   https://git.kernel.org/pub/scm/virt/kvm/kvm.git queue
head:   c36edbda86dd5ceeed2bfa94375286074cee72a4
commit: 4589eecce385c0bc5d3bd065ab850f00ef828356 [97/98] KVM: SVM: Workaround errata#1096 (insn_len maybe zero on SMAP violation)

 vmx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/x86/kvm/vmx/vmx.c
+++ b/arch/x86/kvm/vmx/vmx.c
@@ -7410,7 +7410,7 @@ static int enable_smi_window(struct kvm_
 
 static bool vmx_need_emulation_on_page_fault(struct kvm_vcpu *vcpu)
 {
-	return 0;
+	return false;
 }
 
 static __init int hardware_setup(void)
